python/google/protobuf/internal/well_known_types.py

```diff
@@ -372,6 +397,35 @@ def _NormalizeDuration(self, seconds, nanos):
     self.seconds = seconds
     self.nanos = nanos
 
+  def _comparable_vector(self, obj):
+    # comarison with timedelta not possible with python2 - see https://github.com/protocolbuffers/protobuf/pull/5168
+    return (obj.seconds, obj.nanos)
+
+  # Make sure to normalized timestamps before using the following comparisons.
+  # Comparing non-normalized timestamps is not specified and may give unexpected results.
+  def __eq__(self, other):
+    return self._comparable_vector(self) == self._comparable_vector(other)
+
+  def __ne__(self, other):
+    return self._comparable_vector(self) != self._comparable_vector(other)
+
+  def __lt__(self, other):
+    return self._comparable_vector(self) < self._comparable_vector(other)
+
+  def __le__(self, other):
+    return self._comparable_vector(self) <= self._comparable_vector(other)
+
+  def __gt__(self, other):
+    return self._comparable_vector(self) > self._comparable_vector(other)
+
+  def __ge__(self, other):
+    return self._comparable_vector(self) >= self._comparable_vector(other)
+
+  def __bool__(self):
```

Comment:
This is not really a comparison, and the first time that a Message has a boolean value. Can't users just call `ToTimedelta()`?

Reply:
Good point, I just amended this out